receiver/hostmetricsreceiver/internal/scraper/memoryscraper/factory.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memoryscraper
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/receiver/hostmetricsreceiver/internal"
+)
+
+// This file implements Factory for Memory scraper.
+
+const (
+	// The value of "type" key in configuration.
+	TypeStr = "memory"
+)
+
+// Factory is the Factory for scraper.
+type Factory struct {
+}
+
+// Type gets the type of the scraper config created by this Factory.
+func (f *Factory) Type() string {
+	return TypeStr
+}
+
+// CreateDefaultConfig creates the default configuration for the Scraper.
+func (f *Factory) CreateDefaultConfig() internal.Config {
+	return &Config{}
+}
```

Comment:
Would be good to provide some useful default for this so user can just write the scraper name in the config. Probably same comment applies to cpu, maybe do it in a separate PR?

Reply:
The cpu scraper defines a useful default. The memory scraper doesn't have any config so this is just an empty struct, but the user does only need to write the scraper name in the config (see https://github.com/open-telemetry/opentelemetry-collector/pull/911/files#diff-a9325b7985e2499b8d97815fdbda4a26R11)